### PR TITLE
dev: update plugin for glimmer compatibility

### PR DESCRIPTION
**Pull request by featheredtoast**
Add plugin outlet connectors to attach pure-glimmer implementations of group-tracker buttons and functionality.

This implementation works simultaneously for both widget and glimmer implementations of topic timeline.

This still retains all widget attachments - once widgets are removed from the topic timeline, we can remove widget references here.

Group tracker nav on mobile view - update to glimmer component

This implementation at time of writing hinges on there being plugin outlets for `timeline-controls-before` and `timeline-footer-controls-after`

### CODE REVIEW

File: assets/javascripts/discourse/components/group-tracker-first-post.hbs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{{#if @topic.first_tracked_post}}
+  <DButton
+    class="first-tracked-post"
```
Comment (janzenisaac) [inline assets/javascripts/discourse/components/group-tracker-first-post.hbs:3]:
```suggestion
    @class="first-tracked-post"
```

File: assets/javascripts/discourse/components/group-tracker-first-post.hbs [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    @icon="arrow-circle-up"
+    @title="group_tracker.first_post"
+    @disabled={{this.disabled}}
+           @action={{action "jumpToFirstTrackedPost"}}
```
Comment (janzenisaac) [inline assets/javascripts/discourse/components/group-tracker-first-post.hbs:7]: Can we fix the indentation here?

File: assets/javascripts/discourse/components/group-tracker-nav.hbs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<div class="group-tracker-nav">
+  {{#if this.groupTrackerPostsExist}}
+    <DButton
+      class="btn-default group-tracker-jump-prev"
```
Comment (janzenisaac) [inline assets/javascripts/discourse/components/group-tracker-nav.hbs:4]:
```suggestion
      @class="btn-default group-tracker-jump-prev"
```

File: assets/javascripts/discourse/components/group-tracker-nav.hbs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      @action={{this.jumpToPrevTrackedPost}}
+      @icon={{this.prevTrackerIcon}}
+      @disabled={{this.prevTrackedPostDisabled}}
+    >
+      {{d-icon "arrow-left"}}
+    </DButton>
+    <DButton
+      class="btn-default group-tracker-jump-next"
```
Comment (janzenisaac) [inline assets/javascripts/discourse/components/group-tracker-nav.hbs:12]:
```suggestion
      @class="btn-default group-tracker-jump-next"
```

File: assets/javascripts/discourse/components/group-tracker-nav.js [hunk 5]
```diff
@@ -59,68 +40,72 @@ export default Component.extend({
...
+    return this.prevTrackedPostGroup === null;
+  }
+
+  @action
+  jumpToNextTrackedPost() {
+    const topic = this.args.topic;
+    const nextTrackedPost = this.getNextTrackedPost();
+
+    if (nextTrackedPost) {
+      const url = topic.url + "/" + nextTrackedPost.post_number;
```
Comment (janzenisaac) [inline assets/javascripts/discourse/components/group-tracker-nav.js:96]:
```suggestion
      const url = this.args.topic.url + "/" + nextTrackedPost.post_number;
```

Comment (janzenisaac) [reply assets/javascripts/discourse/components/group-tracker-nav.js:96]: and you can drop that `const` assignment above

File: assets/javascripts/discourse/components/group-tracker-nav.js [hunk 6]
```diff
@@ -59,68 +40,72 @@ export default Component.extend({
...
+    }
+  }
+
+  @action
+  jumpToPrevTrackedPost() {
+    const topic = this.args.topic;
+    const prevTrackedPost = this.getPreviousTrackedPost();
+
+    if (prevTrackedPost) {
+      const url = topic.url + "/" + prevTrackedPost.post_number;
```
Comment (janzenisaac) [inline assets/javascripts/discourse/components/group-tracker-nav.js:107]:
```suggestion
      const url = this.args.topic.url + "/" + prevTrackedPost.post_number;
```

same as above.

File: assets/javascripts/discourse/components/group-tracker-first-post.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import { action } from "@ember/object";
+import Component from "@glimmer/component";
+import DiscourseURL from "discourse/lib/url";
+import { alias } from "@ember/object/computed";
+
+export default class GroupTrackerFirstPost extends Component {
+  @alias("args.topic.currentPost") postId;
```
Comment (janzenisaac) [inline assets/javascripts/discourse/components/group-tracker-first-post.js:7]: A bit of a nitpick, but I think doing something like

```suggestion
  postId = this.args.topic.currentPost
```

feels more _glimmery_. What do you think? If you are cool with the change there are couple instances of this.

**Review by janzenisaac — COMMENTED** [review]
Looks great, a couple of thoughts for you :)

File: assets/javascripts/discourse/components/group-tracker-first-post.js (hunk 7 above)
Comment (CvX) [reply assets/javascripts/discourse/components/group-tracker-first-post.js:7]: `postId = this.args.topic.currentPost` wouldn't be reactive to potential args changes. I'd recommend `@readOnly` over `@alias` though

File: assets/javascripts/discourse/components/group-tracker-first-post.hbs (hunk 1 above)
Comment (featheredtoast) [reply assets/javascripts/discourse/components/group-tracker-first-post.hbs:3]: I understand that non-`@` attributes can get passed directly as rendered HTML attributes - we're keeping `@class` support in DButton for backwards compatibility (for now), see https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/app/templates/components/d-button.hbs#L3

**Comment by featheredtoast** [comment]
@janzenisaac rebased on lates, and committed most of the suggestions.

(Note the comment regarding `@class` - that's still supported in DButton, but recommended is use `class` as native attribute)

File: assets/javascripts/discourse/components/group-tracker-first-post.hbs (hunk 2 above)
Comment (featheredtoast) [reply assets/javascripts/discourse/components/group-tracker-first-post.hbs]: Done - looks like my editor needs a prettier update, too.

File: assets/javascripts/discourse/components/group-tracker-first-post.hbs (hunk 1 above)
Comment (janzenisaac) [reply assets/javascripts/discourse/components/group-tracker-first-post.hbs:3]: Good to know, thanks for pointing this out :)

**Review by janzenisaac — APPROVED** [review]

**Pull request merged**